components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/ManagementApiParser.java

```diff
@@ -0,0 +1,148 @@
+ *   under the License.
+ */
+package org.wso2.micro.integrator.management.apis;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.micro.core.util.CarbonException;
+import org.wso2.micro.integrator.core.util.MicroIntegratorBaseUtils;
+import org.wso2.securevault.SecretResolverFactory;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * This class reads through internal-apis.xml and parses the values.
+ */
+public class ManagementApiParser {
+
+    private static final Log LOG = LogFactory.getLog(ManagementApiParser.class);
+    private static OMElement managementApiElement;
+
+    public ManagementApiParser() {
+    }
+
+    /**
+     * Method to get the File object representation of the internal-apis.xml file.
+     *
+     * @return File object representation
+     */
+    public static File getConfigurationFile() {
+        return new File(MicroIntegratorBaseUtils.getCarbonConfigDirPath(), "internal-apis.xml");
+    }
+
+    /**
+     * Method to get absolute path of the internal-apis.xml file.
+     *
+     * @return The absolute path for the file
+     */
+    public static String getConfigurationFilePath() {
+        return new File(MicroIntegratorBaseUtils.getCarbonConfigDirPath(), "internal-apis.xml").getAbsolutePath();
+    }
+
+    public static OMElement getManagementApiElement() throws IOException, CarbonException, XMLStreamException,
+            ManagementApiUndefinedException {
+        if (Objects.nonNull(managementApiElement)) {
+            return managementApiElement;
+        }
+        LOG.debug("Parsing " + getConfigurationFilePath() + " for the first time.");
+        Iterator<OMElement> internalApis = getInternalApisElement();
+        while (internalApis.hasNext()) {
+            OMElement apiOM = internalApis.next();
+            String apiName = apiOM.getAttributeValue(new QName("name"));
+            if ("ManagementApi".equals(apiName)) {
+                managementApiElement = apiOM;
+                return apiOM;
+            }
+        }
+        throw new ManagementApiUndefinedException("Management API not defined in " + getConfigurationFilePath());
+    }
+
+    public HashMap<String, char[]> getUserList() throws CarbonException, XMLStreamException, IOException,
+            UserStoreUndefinedException, ManagementApiUndefinedException {
+        OMElement userStoreOM = getManagementApiElement().getFirstChildWithName(new QName("UserStore"));
+        if (Objects.nonNull(userStoreOM)) {
+            return populateUserList(userStoreOM.getFirstChildWithName(new QName("users")));
```

Comment:
Shall we hash this value and return, though we are initializing only once form SecurityHandlerAdapter in case if some other method uses this in future it will reduce the reads.

Reply:
done

Reply:
I couldn't find an assignment for the hashed value.

Reply:
done